src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<T> producer = null;
+        if (type == Bulldozer.class) {
+            producer = (MachineProducer<T>) new BulldozerProducer();
+        }
+        if (type == Truck.class) {
+            producer = (MachineProducer<T>) new TruckProducer();
+        }
+        if (type == Excavator.class) {
+            producer = (MachineProducer<T>) new ExcavatorProducer();
+        }
+        if (producer != null) {
+            return (List<T>) producer.get();
+        } else {
+            return List.of();
```

Comment:
Consider creating a local variable for the machines list. This way, you can avoid casting and directly return the result from the producer. For example: 'List<T> machines = producer.get(); return new ArrayList<>(machines);'